src/main/java/org/gaul/s3proxy/TierBlobStore.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2014-2021 Andrew Gaul <andrew@gaul.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gaul.s3proxy;
+
+import static java.util.Objects.requireNonNull;
+
+import org.jclouds.blobstore.BlobStore;
+import org.jclouds.blobstore.domain.Blob;
+import org.jclouds.blobstore.domain.BlobMetadata;
+import org.jclouds.blobstore.domain.MultipartUpload;
+import org.jclouds.blobstore.domain.Tier;
+import org.jclouds.blobstore.domain.internal.BlobMetadataImpl;
+import org.jclouds.blobstore.options.PutOptions;
+import org.jclouds.blobstore.util.ForwardingBlobStore;
+
+/**
+ * This class implements a middleware to set the storage tier when creating
+ * objects.  The class is configured via:
+ *
+ *   s3proxy.tier-blobstore = VALUE
+ *
+ * VALUE can be anything from org.jclouds.blobstore.domain.Tier, e.g.,
+ * STANRDARD, COOL, COLD, ARCHIVE.  org.jclouds.s3.domain.StorageClass
```

Comment:
If I use COOL in the configuration 
```
s3proxy.tier-blobstore=COOL
```

I get this error message 
```
Exception in thread "main" java.lang.IllegalArgumentException: No enum constant org.jclouds.blobstore.domain.Tier.COOL
        at java.base/java.lang.Enum.valueOf(Enum.java:293)
        at org.jclouds.blobstore.domain.Tier.valueOf(Tier.java:24)
        at org.gaul.s3proxy.Main.parseMiddlewareProperties(Main.java:278)
        at org.gaul.s3proxy.Main.main(Main.java:125)
```

Look like we are limited by the enum values of [org.jclouds.blobstore.domain.Tier](https://jclouds.apache.org/reference/javadoc/2.1.x/org/jclouds/blobstore/domain/Tier.html) which are 

```
STANDARD
INFREQUENT
ARCHIVE
```

I would suggest to change this comment accordingly unless I pick up the wrong version of jcloud in the maven dependencies.

Reply:
Most likely that's what happened in this version there is COOL and COLD
https://github.com/apache/jclouds/blob/master/blobstore/src/main/java/org/jclouds/blobstore/domain/Tier.java
I did not figure out why when I built s3proxy but that does not matter please ignore my comment.

Reply:
jclouds 2.6.0 only has a few Tier constants but 2.7.0-SNAPSHOT includes a wider variety.